cli/pcluster/utils.py

```diff
@@ -113,7 +113,8 @@ def get_region():
 def get_partition():
     """Get partition for the AWS_DEFAULT_REGION set in the environment."""
     region = get_region()
-    return next(("aws-" + partition for partition in ["us-gov", "cn"] if region.startswith(partition)), "aws")
+    if region:
```

Comment:
instead of adding this check let's change the `get_region` function to raise an exception if there is no region in the environment.

All the code relies on the fact we have a region in the enviornment. If it's not true we should fail.